src/herder/TransactionQueue.cpp

```diff
@@ -116,10 +116,10 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx,
     //
     // FEE_MULTIPLIER * v2 does not overflow uint128_t because fees are bounded
     // by INT64_MAX, while number of operations and FEE_MULTIPLIER are small.
-    uint128_t v1 = bigMultiply(newFee, oldNumOps);
     uint128_t v2 = bigMultiply(oldFee, newNumOps);
     uint128_t minFeeN = v2 * TransactionQueue::FEE_MULTIPLIER;
-    bool res = v1 >= minFeeN;
+
+    bool res = newFee >= 0 && bigMultiply(newFee, oldNumOps) >= minFeeN;
```

Comment:
This check is good and makes sense to have here. But I would also add a simple negative inclusion fee check before adding tx to the limiter in the first place (IIUC that's `canAdd` method in tx queue).

Reply:
Added a check close to the top of `canAdd`.